pkg/cmd/pulumi/packageresolution/package_resolution.go

```diff
@@ -65,127 +68,332 @@ func (e PackageNotFoundError) Suggestions() []apitype.PackageMetadata {
 }
 
 type Options struct {
-	DisableRegistryResolve      bool
-	Experimental                bool
-	IncludeInstalledInWorkspace bool
+	// If [Resolve] should use the passed in registry to resolve packages.
+	ResolveWithRegistry bool
+
+	// If the resolution should use already installed plugins when resolving
+	// plugins without specific versions provided.
+	//
+	// Concretely, when resolving a package like aws, if a version of aws is
+	// already on disk, then that version will be preferred over latest.
+	ResolveVersionWithLocalWorkspace bool
+
+	// Resolve the source directly against the local workspace if possible.
+	//
+	// For example, consider a spec like workspace.PackageSpec{Source:
+	// "example"}. With "AllowNonInvertableLocalWorkspaceResolution: false", resolving
+	// "example" may resolve to a parameterized package with a base plugin called
+	// "base". [Resolve] wouldn't consider a local plugin called "example", and might
+	// fail if "base" couldn't be installed.
+	//
+	// "AllowNonInvertableLocalWorkspaceResolution: true" instructs [Resolve] to use a
+	// package called "example" if example is requested and present in the workspace,
+	// regardless of external lookup.
+	AllowNonInvertableLocalWorkspaceResolution bool
```

Comment:
I am still struggling with this. From what I understand this is only needed when the registry is in play, right? Does this flag basically mean: prioritise already installed plugins over what the registry would tell you? If so, why do we need this behaviour?